src/classes/ALLO_Multicurrency_TDTM.cls

```diff
@@ -159,33 +156,14 @@ public class ALLO_Multicurrency_TDTM extends TDTM_Runnable {
         return results;
     }
 
-    /*******************************************************************************************************
-    * @description Detects if this organization has multicurrency enabled, and populates class variables if 
-    * so.
-    * @return boolean Whether this org has multicurrency enabled.
-    ********************************************************************************************************/
-    public static boolean isMultiCurrency() {
-        if (multiCurrency==null) {
-
-            String queryExchangeRates = 'select IsoCode, ConversionRate, IsCorporate from CurrencyType';
-            SObjectType soCurrencyType = Schema.getGlobalDescribe().get('CurrencyType'); 
-
-            //are we in a multi-currency org?
-            if (userInfo.isMultiCurrencyOrganization() && soCurrencyType != null) {
-                multiCurrency = true;
-
-                //iterate over all the currencies in the org (inc. inactive ones as they may have been used previously)
-                //this allows us to put them in a map and use the ISO code as key and also set the corporate
-                for(sObject ct: Database.query(queryExchangeRates)){
-                    exchangeRates.put((string)ct.get('IsoCode'),(decimal)ct.get('ConversionRate'));
-                    if ((boolean)ct.get('IsCorporate'))
-                        currCorporate=(string)ct.get('IsoCode');
-                }
-            }
-            else
-                multiCurrency = false; 
-        }        
-        return multiCurrency;
+
+    /**
+     * @description Wrapper for the utility method to determine if multi-currency is enabled in the org
+     * The target utility class can be mocked for unit tests.
+     * @return True if multi-currency is enabled in the org
+     */
+    public static Boolean isMultiCurrency() {
+        return UTIL_Currency.getInstance().isMultiCurrencyOrganization();
```

Comment:
Is this method necessary at all?

Reply:
Let's remove all these different methods and consolidate them all to use UserInfo.isMultiCurrencyOrganization()

Reply:
They do have slightly different uses though:
* UserInfo.isMultiCurrencyOrganization() - Absolute. Based on the Org's configuration.
* UTIL_Currency.getInstance().isMultiCurrencyOrganization() - Can be mocked in Unit Tests to return a value of True even if the org doesn't support multi-currency.

Reply:
Makes sense!